ledfx/api/websocket.py

```diff
@@ -297,6 +300,36 @@ def audio_stream_data_handler(self, message):
             message.get("data").values(), dtype=np.float32
         )
 
+    @websocket_handler("audio_stream_data_v2")
+    def audio_stream_data_base64_handler(self, message):
+        # Max value for signed 16-bit values.
+        MAX_VAL = 32767
+        if not ACTIVE_AUDIO_STREAM:
+            return
+
+        client = message.get("client")
+
+        if ACTIVE_AUDIO_STREAM.client != client:
+            return
+        try:
+            decoded = pybase64.b64decode(message.get("data"))
+        except binascii.Error:
+            _LOGGER.info("Incorrect base64 padding.")
+        except Exception as err:
+            _LOGGER.exception(
+                "Unexpected Exception in base64 decoding: %s", err
+            )
+        else:
+            fmt = "<%dh" % (len(decoded) // 2)
+            data = list(struct.unpack(fmt, decoded))
+            # Minimum value is -32768 for signed, so that's why if the number is negative,
+            # it is divided by 32768 when converting to float.
+            data = np.array(
+                [d / MAX_VAL if d >= 0 else d / (MAX_VAL + 1) for d in data],
+                dtype=np.float32,
+            )
```

Comment:
Copilot also suggests this, presumably it's similar to what you've benchmarked.
`data = np.where(decoded >= 0, decoded / MAX_VAL, decoded / (MAX_VAL + 1)).astype(np.float32)`

Reply:
This doesn't work for two reasons:

1. Python complains about comparison being done between int and byte.
2. This goes through each byte (which is 8 bits), dividing it by the max value on a signed 16 bit integer. The data with webaudio being received is 16 bits, therefore, two bytes need to be taken and then unpacked to a 16 bit integer.

The right option in this case when using numpy is to use `data = np.frombuffer(decoded, dtype=np.int16)`, however, this is at least two times slower than the method I am using here. Tested this on multiple machines, all showed the same drop in performance when using numpy.